memory/ndctl.py

```diff
@@ -418,6 +419,57 @@ def test_check_namespace(self):
                           ignore_status=True):
             self.fail("Failed to check namespace metadata")
 
+    def test_check_numa(self):
+        self.enable_region()
+        regions = self.get_json(short_opt='-R')
+        for reg in regions:
+            numa = genio.read_one_line('/sys/devices/ndbus%s/%s/numa_node'
+                                       % (re.findall(r'\d+', reg)[0], reg))
+            # Check numa config in ndctl and sys interface
+            if len(self.get_json(long_opt='-r %s -U %s' % (reg, numa))) != 1:
+                self.fail('Region mismatch between ndctl and sys interface')
+
+    def test_label_read_write(self):
+        self.enable_region()
+        region = self.params.get('region', default=None)
+        if not region:
+            region = self.get_json(short_opt='-R')[0]
+        nmem = "nmem%s" % re.findall(r'\d+', region)[0]
+
+        self.log.info("Using %s for testing labels", region)
+        self.disable_region(name=region)
+        self.log.info("Filling zeros to start test")
+        if process.system('%s zero-labels %s' % (self.binary, nmem), shell=True):
+            self.fail("Label zero-fill failed")
+
+        self.enable_region(name=region)
+        self.create_namespace(region=region)
+        self.log.info("Storing labels with a namespace")
+        old_op = process.system_output('%s check-labels %s' % (self.binary, nmem), shell=True)
+        if process.system('%s read-labels %s -o output' % (self.binary, nmem), shell=True):
+            self.fail("Label read failed")
+
+        self.log.info("Refilling zeroes before a restore")
+        self.disable_namespace(region=region)
+        self.destroy_namespace(region=region)
+        self.disable_region(name=region)
+        if process.system('%s zero-labels %s' % (self.binary, nmem), shell=True):
+            self.fail("Label zero-fill failed after read")
+
+        self.log.info("Re-storing labels with a namespace")
+        if process.system('%s write-labels %s -i output' % (self.binary, nmem), shell=True):
+            self.fail("Label write failed")
+        self.enable_region(name=region)
+
+        self.log.info("Checking mismatch after restore")
+        new_op = process.system_output('%s check-labels %s' % (self.binary, nmem), shell=True)
+        if new_op != old_op:
+            self.fail("Label read and write mismatch")
```

Comment:
@harish-24 can we have a function which can be called twice (before and after destroying namespace), instead of repeating code for some of the operations ?

Reply:
@narasimhan-v we have the following operations.
```
disable-region
zero-labels
enable-region
create
check-labels
read-labels
disable-namespace
destroy-namespace
disable-region
zero-labels
write-labels
enable-region
check-labels
```
you mean to have functions for individual tasks and call in the same order as above?